client/src/lib.rs

```diff
@@ -0,0 +1,143 @@
+//! Client libarary for the job runner
+
+use std::convert::TryInto;
+
+use tokio::sync::mpsc::Sender;
+use tokio::time::{sleep, Duration};
+use tonic::transport::{Certificate, Channel, ClientTlsConfig, Identity};
+
+use common::t_service_client::TServiceClient;
+use common::*;
+
+// Re-exports
+pub use common::output_event::Stream as OutputStream;
+pub use common::{JobId, JobStartRequest};
+
+#[derive(Debug, Clone)]
+pub struct TlsConfig {
+    /// Contents of the PEM-formatted server root CA certificate file
+    pub server_root_ca_crt: Vec<u8>,
+    /// Contents of the PEM-formatted client certificate file
+    pub client_crt: Vec<u8>,
+    /// Contents of the PEM-formatted client private key file
+    pub client_key: Vec<u8>,
+}
+
+// TODO: create a proper error type, and add conversions to it
+
+type DResult<T> = Result<T, Box<dyn std::error::Error>>;
+
+pub struct Client {
+    client: TServiceClient<Channel>,
+}
+impl Client {
+    /// Connect to a job server
+    pub async fn connect(target: String, tls: TlsConfig) -> DResult<Self> {
```

Comment:
This should take &str if possible.

Reply:
The internal client requires owned `String` anyway, so I'm going to use `Into<String>` to accept both. Fixed in 6304652.